drivers/kscan/kscan_xpt2046.c

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2023 Seppo Takalo
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/kscan.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(xpt2046, CONFIG_KSCAN_LOG_LEVEL);
+
+struct xpt2046_config {
+	const struct spi_dt_spec bus;
+	const struct gpio_dt_spec int_gpio;
+	uint16_t min_x;
+	uint16_t min_y;
+	uint16_t max_x;
+	uint16_t max_y;
+	uint16_t threshold;
+	uint16_t screen_size_x;
+	uint16_t screen_size_y;
+	uint16_t reads;
+};
+struct xpt2046_data {
+	const struct device *dev;
+	kscan_callback_t callback;
+	struct gpio_callback int_gpio_cb;
+	bool enabled;
+	struct k_work work;
+	struct k_work_delayable dwork;
+	uint8_t rbuf[9];
+	uint32_t last_x;
+	uint32_t last_y;
+	bool pressed;
+};
+
+enum xpt2046_channel {
+	CH_TEMP0 = 0,
+	CH_Y,
+	CH_VBAT,
+	CH_Z1,
+	CH_Z2,
+	CH_X,
+	CH_AUXIN,
+	CH_TEMP1
+};
+
+struct measurement {
+	uint32_t x;
+	uint32_t y;
+	uint32_t z;
+};
+
+#define START		      BIT(7)
+#define CHANNEL(ch)	      ((ch & 0x7) << 4)
+#define MODE_8_BIT	      BIT(3)
+#define SINGLE_ENDED	      BIT(2)
+#define POWER_OFF	      0
+#define POWER_ON	      0x03
+#define CONVERT_U16(buf, idx) ((uint16_t)((buf[idx] & 0x7f) << 5) | (buf[idx + 1] >> 3))
+
+/* Read all Z1, X, Y, Z2 channels using 16 Clocks-per-Conversion mode.
+ * See the manual https://www.waveshare.com/w/upload/9/98/XPT2046-EN.pdf for details.
+ * Each follow-up command interleaves with previous conversion.
+ * So first command starts at byte 0. Second command starts at byte 2.
+ */
+static uint8_t tbuf[9] = {
+	[0] = START | CHANNEL(CH_Z1) | POWER_ON,
+	[2] = START | CHANNEL(CH_Z2) | POWER_ON,
+	[4] = START | CHANNEL(CH_X) | POWER_ON,
+	[6] = START | CHANNEL(CH_Y) | POWER_OFF,
+};
+
+static void xpt2046_isr_handler(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	struct xpt2046_data *data = CONTAINER_OF(cb, struct xpt2046_data, int_gpio_cb);
+	const struct xpt2046_config *config = data->dev->config;
+
+	gpio_remove_callback(config->int_gpio.port, &data->int_gpio_cb);
+	k_work_submit(&data->work);
+}
+
+static int xpt2046_read_and_cumulate(const struct spi_dt_spec *bus, const struct spi_buf_set *tx,
+				     const struct spi_buf_set *rx, struct measurement *meas)
+{
+	int ret = spi_transceive_dt(bus, tx, rx);
+
+	if (ret) {
```

Comment:
You are mixing the `if (ret)` with `if (ret <0)`. Please unify it (personally I prefer `if (ret < 0)` over `if (ret)` but as I know there is no guideline for that in Zephyr).

Reply:
Fixed.